### PR TITLE
Ensure yaml is handling large Integers comming from git commit IDs co…

### PR DESCRIPTION
**Pull request by ashahba**
…rrectly during Helm install

### CODE REVIEW

**Comment by ashahba** [comment]
/

**Comment by ashahba** [comment]
There is a better fix for this to just pass `--set-string` as described in this ticket:
https://github.com/kubernetes/helm/issues/1707

I'll get a PR for docs update.

**Comment by ashahba** [comment]
For example here is problematic `git` SHA1: `3806521ad609a325c852615ce1f04e6a2aea1c53` that leads to tag: `3806521` which is problematic:

```
$ git checkout 3806521ad609a325c852615ce1f04e6a2aea1c53
$ YOUR_NAMESPACE=ashahba
$ helm install helm-charts/kube-volume-controller -n kvc-$YOUR_NAMESPACE --wait --namespace=$YOUR_NAMESPACE --set tag="$(git rev-parse --short HEAD)" --set clusterrole.install=false --set storageclass.install=false --set crd.install=false --set log_level=4 --set namespace=$YOUR_NAMESPACE
Error: release kvc-ashahba failed: timed out waiting for the condition
$ kubectl get pods
NAME                                                    READY     STATUS              RESTARTS   AGE
kvc-74b69d5c46-v62lp                                    0/1       InvalidImageName    0          6m
$ kubectl logs kvc-74b69d5c46-v62lp -f
Error from server (BadRequest): container "kvc" in pod "kvc-74b69d5c46-v62lp" is waiting to start: InvalidImageName
$ kubectl describe pod kvc-74b69d5c46-v62lp
Name:               kvc-74b69d5c46-v62lp
Namespace:          ashahba
Priority:           0
PriorityClassName:  <none>
Node:               gke-ashahba-kvc-default-pool-1e30c50a-n9pl/10.138.0.13
Start Time:         Wed, 16 May 2018 22:47:15 -0700
Labels:             app=kvc
                    pod-template-hash=3062581702
                    servicetype=controller
Annotations:        <none>
Status:             Pending
IP:                 10.48.0.114
Controlled By:      ReplicaSet/kvc-74b69d5c46
Containers:
  kvc:
    Container ID:  
    Image:         volumecontroller/kube-volume-controller:3.806521e+06
    Image ID:      
    Port:          <none>
    Command:
      /kvc
      --podFile=/kvc-templates/pod.tmpl
      --pvFile=/kvc-templates/pv.tmpl
      --pvcFile=/kvc-templates/pvc.tmpl
      --v=4
      --namespace=ashahba
    State:          Waiting
      Reason:       InvalidImageName
    Ready:          False
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kvc-token-hj4xz (ro)
Conditions:
  Type           Status
  Initialized    True 
  Ready          False 
  PodScheduled   True 
Volumes:
  kvc-token-hj4xz:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  kvc-token-hj4xz
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason                 Age                From                                                 Message
  ----     ------                 ----               ----                                                 -------
  Normal   Scheduled              10m                default-scheduler                                    Successfully assigned kvc-74b69d5c46-v62lp to gke-ashahba-kvc-default-pool-1e30c50a-n9pl
  Normal   SuccessfulMountVolume  10m                kubelet, gke-ashahba-kvc-default-pool-1e30c50a-n9pl  MountVolume.SetUp succeeded for volume "kvc-token-hj4xz"
  Normal   SandboxChanged         10m (x2 over 10m)  kubelet, gke-ashahba-kvc-default-pool-1e30c50a-n9pl  Pod sandbox changed, it will be killed and re-created.
  Warning  InspectFailed          8m (x11 over 10m)  kubelet, gke-ashahba-kvc-default-pool-1e30c50a-n9pl  Failed to apply default image tag "volumecontroller/kube-volume-controller:3.806521e+06": couldn't parse image reference "volumecontroller/kube-volume-controller:3.806521e+06": invalid reference format
  Warning  Failed                 4m (x28 over 10m)  kubelet, gke-ashahba-kvc-default-pool-1e30c50a-n9pl  Error: InvalidImageName
```
Sure enough image tag in ```Image:         volumecontroller/kube-volume-controller:3.806521e+06``` have been parsed as `Float64` instead of `String`!

File: docs/ops.md [hunk 1]
```diff
@@ -111,6 +112,12 @@ $ helm install helm-charts/kube-volume-controller/ -n kvc --wait \
   --set namespace=<kvc_namespace>
 ```
 
+#### Installing KVC from tip of a branch
+You can also install `KVC` from feature branches if needed. In this case follow same steps as described above, but set the `tag` option this way:
+```
+--set-string tag="$(git rev-parse --short HEAD)"
```
Comment (balajismaniam) [inline docs/ops.md]: Why not use `git describe --tags --always --dirty` here as well?

**Comment by ashahba** [comment]
That doesn't work either.

**Comment by ashahba** [comment]
And here was the final fix on the `helm` thead:
https://github.com/kubernetes/helm/pull/3599

**Comment by balajismaniam** [comment]
@ashahba As I mentioned to you offline during our face-to-face meeting in OR, the helm version needs to be updated. Please update that as well and I will merge this PR.

Also, please e2e test it manually before submitting PRs.

**Comment by balajismaniam** [comment]
@ashahba The e2e tests are still failing. 
Can you please manually test before submitting your PRs?

**Comment by ashahba** [comment]
I can't manually test on CircleCi.
It seems like it's the config now needs `helm` update too.
I'll do it shortly.

**Comment by ashahba** [comment]
@balajismaniam we have to also run `helm init --upgrade` against the cluster and from a client that runs `helm v2.9.0` in order to update `tiller` to this version otherwise we may run into mismatched versions.

I can do it in the PR and within `.circleci/config.yml` but I don't think we want that, because we don't do the initial `helm init` command here.
Let me you what you think.

**Comment by ashahba** [comment]
Ideally we should have:
```
$ helm version
Client: &version.Version{SemVer:"v2.9.0", GitCommit:"f6025bb9ee7daf9fee0026541c90a6f557a3e0bc", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.9.0", GitCommit:"f6025bb9ee7daf9fee0026541c90a6f557a3e0bc", GitTreeState:"clean"}
```

**Comment by balajismaniam** [comment]
Closing due to inactivity. Please reopen if required.

**Pull request closed without merging**